### ISSUE TITLE
input/output error on google drive mount with filename longer than 1024 chars

### ISSUE THREAD

**Issue by FlorianLudwig** [rclone/rclone#3884]
#### What is the problem you are having with rclone?
Receiving `Input/output error`s. I use it regularly and didn't have this issue in the past but I did not change anything knowingly.

#### What is your rclone version (output from `rclone version`)
I tried several, including the latest beta:
```
./rclone version
rclone v1.50.2-146-g3a1b41ac-beta
- os/arch: linux/amd64
- go version: go1.13.6
```

#### Which OS you are using and how many bits (eg Windows 7, 64 bit)
Linux, 64 bit

####  Which cloud storage system are you using? (eg Google Drive)
Google Drive

#### The command you were trying to run (eg `rclone copy /tmp remote:tmp`)
```
rclone mount
```

#### A log from the command with the `-vv` flag (eg output from `rclone -vv copy /tmp remote:tmp`)
I mounted with `--log-level DEBUG --dir-cache-time 60m` and then run `find gdrivefs/ -maxdepth 1` twice. Both find commands end with `find: ‘gdrivefs/’: Input/output error`.

The rclone command produces a ton of output but nothing looking suspicious. On the second run with everything cached it only procudes:

```
2020/01/16 19:49:30 DEBUG : /: ReadDirAll: 
2020/01/16 19:49:30 DEBUG : /: >ReadDirAll: item=1556, err=<nil>
```

**Comment by ncw** [comment]
Can you run the find command again with `--debug-fuse` which will create even more output and post the result please?  Attach it to this issue would be ideal, or email it to me nick@craig-wood.com with a link to this issue in if it has confidential stuff in.

Are there any versions of rclone which do work OK?

**Comment by animosity22** [comment]
How you are mounting it? Just running it from the command line and leaving it in the foreground / systemd service / some other method?

The input / output error usually is around the process being killed/dying/disappearing and fuse doesn't think it has been unmounted so the fuse part remains as something is accessing the mount point.

Are you able to share the whole debug log?

**Comment by FlorianLudwig** [comment]
> Are there any versions of rclone which do work OK?
No

> How you are mounting it? Just running it from the command line and leaving it in the foreground / systemd service / some other method?
I start it on the command line in foreground. It does not die.

> Are you able to share the whole debug log?
Sadly no (does contain PII of not just me but colleagues). I will continue to try narrow down how to reproduce a minimal log. I did try to `strace find` to see which call exactly does produce the error but no luck so far.

**Comment by animosity22** [comment]
So you are getting an input/output error and the process is still there? 

What OS / kernel?

```
felix@gemini:~$ lsb_release -a
No LSB modules are available.
Distributor ID:	Debian
Description:	Debian GNU/Linux 10 (buster)
Release:	10
Codename:	buster
felix@gemini:~$ uname -a
Linux gemini 4.19.0-6-amd64 #1 SMP Debian 4.19.67-2+deb10u2 (2019-11-11) x86_64 GNU/Linux
felix@gemini:~$
```

**Comment by FlorianLudwig** [comment]
```

**Comment by animosity22** [comment]
Can you share the kernel as well? 

```
uname -a
```

**Comment by FlorianLudwig** [comment]
> So you are getting an input/output error and the process is still there?

Yes. I also made sure that the log:

```
2020/01/16 19:49:30 DEBUG : /: ReadDirAll: 
2020/01/16 19:49:30 DEBUG : /: >ReadDirAll: item=1556, err=<nil>
```

is exactly what happened during the `find` (with the data cached), nothing else. I do have ~ 1.5k files in the folder. I am now moving some away and see if it still happens.

> Kernel

```
$ uname -a
Linux knight1 5.3.16-300.fc31.x86_64 #1 SMP Fri Dec 13 17:59:04 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by animosity22** [comment]
I did on a test VM but the kernel version is ahead of yours.

![image](https://user-images.githubusercontent.com/5088712/72644682-9a241400-393f-11ea-8309-7cec930fc19b.png)

I'm also not using the beta version but the latest stable. Are you using the beta for a particular fix? Does the same issue happen on 1.50.2?

**Comment by FlorianLudwig** [comment]
I tried the beta because of this issue. I tested two stable releases first though.

I had 1.5k files in the root of my drive (and then more in folders). I moved all files that where directly in the root folder to a new directory and left all the others folders as they where. Now I get the error if I list the files in that new folder. Narrowed it down somewhat ;) so if is not related to folders/subfolders.

**Comment by FlorianLudwig** [comment]
interesting fact: calling `find` results in an input/output error. `ls` works fine.

**Comment by ncw** [comment]
Can you run with `--debug-fuse` and just examine the logs for the second (cached) run?

It must be some interaction with fuse that isn't touching rclone.  Maybe there is a file with an incorrect name (eg contains a `/` or is `.` or `..`) that the kernel doesn't like.

You should see the error which will look something like `error=EIO` hopefully.

If you can post the bits before that (redact as necessary) that should help!

**Comment by FlorianLudwig** [comment]
After some more bisecting: I have a file name of 1040 characters. It causes the error.

**Comment by FlorianLudwig** [comment]
How to reproduce:
```
aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaab
```

Would produce the error (1025 chars). Removing the `b` at the end, gets it back down to 1024 and it works again.

**Comment by ncw** [comment]
Well done for working the problem out :-)

I think we had a similar problem before - see 59026c4761e3926f6a54b6518472c54345e0b7fc with length 4k.  I wonder if FUSE has changed since then.

I can change the threshold to 1k so these files get skipped easily enough

I've done this here

https://beta.rclone.org/branch/v1.50.2-171-g1f071956-fix-3884-long-name-beta/ (uploaded in 15-30 mins)

Can you have a go?

I'm not 100% happy with just dropping the files (with an ERROR message) though.

**Comment by ncw** [comment]
Did that work?

**Comment by ncw** [comment]
I've got a patch pending to fix this - but I'd like to hear from you @FlorianLudwig to see if it fixes your problem or not. Can you have a test of the above beta code?

Thanks

**Comment by animosity22** [comment]
Hmm. I can't seem to replicate it as all my file systems are not letting me make a file that long as I just get file too long.

@FlorianLudwig  - are you able to give a test so we can confirm it works?

**Comment by FlorianLudwig** [comment]
@animosity22 @ncw To reproduce the problem:

1. create an empty document on google drive
2. name it: "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
3. mount drive with rclone
4. `ls`

Without the patch just "I/O error" appears. With the patch:

```
2020/05/21 16:20:13 ERROR : /: Name too long (1034 bytes) for FUSE, skipping: aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa.odt
```

I am happy with that. Mapping filenames might lead to other problems... At least now it's clear what's wrong, thanks!

**Comment by animosity22** [comment]
Thanks. I was able to reproduce the issue and get the I/O error.

I was able to test the patch and it skips the file and generates an error:

```
2020/05/21 10:30:12 ERROR : /: Name too long (1036 bytes) for FUSE, skipping: aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaab.docx
```

I think I'm good with that result.

**Comment by ncw** [comment]
Thanks for testing :-)

I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org) in 15-30 mins and released in v1.52
